fuel-vm/Cargo.toml

```diff
@@ -36,7 +36,7 @@ tai64 = "4.0"
 thiserror = "1.0"
 
 [dev-dependencies]
-ed25519-dalek = "1.0" # TODO: upgrade to 2.0 when it's released, and remove rand below
+ed25519-dalek = "2.0.0"
 ed25519-dalek_old_rand = { package = "rand", version = "0.7.3" }
```

Comment:
Do we still need to use this one?=)

Reply:
<img width="646" alt="image" src="https://github.com/FuelLabs/fuel-vm/assets/18346821/a5f57ee8-1ff7-45f9-bc0d-fce49c58a432">

It has `rand_core` feature, maybe we can use it isntead

Reply:
Yep. That seems to work.